Source/WebCore/platform/graphics/gbm/MemoryMappedGPUBuffer.cpp

```diff
@@ -0,0 +1,505 @@
+                LOG(Images, " -> buffer allocated without modifiers (gbm_bo reports modifier 0x%2lx, m_modifier 0x%2lx)", gbm_bo_get_modifier(m_bo), m_modifier);
+        }
+    }
+
+    if (m_bo) {
+        if (!exportGBMBufferObjectAsDMABuf())
+            return false;
+
+        if (dumpBufferAllocation)
+            LOG(Images, " -> exporting gbm_bo buffer handle %u as dma-buf, got fd %i! succeeded!", gbm_bo_get_handle(m_bo).u32, m_dmaBufFD);
+
+        return true;
+    }
+
+    return false;
+}
+
+bool MemoryMappedGPUBuffer::map()
+{
+    ASSERT(!m_mappedData);
+    ASSERT(!m_mappedLength);
+    m_stride = gbm_bo_get_stride_for_plane(m_bo, 0);
+    m_mappedLength = m_stride * m_size.height();
+    m_mappedData = mmap(nullptr, m_mappedLength, PROT_READ | PROT_WRITE, MAP_SHARED, m_dmaBufFD, 0);
+    return m_mappedData != MAP_FAILED;
+}
+
+void MemoryMappedGPUBuffer::unmap()
+{
+    ASSERT(m_mappedData);
+    ASSERT(m_mappedLength > 0);
+    munmap(m_mappedData, m_mappedLength);
+}
+
+bool MemoryMappedGPUBuffer::createEGLImageFromDMABuf()
+{
+    auto planeCount = gbm_bo_get_plane_count(m_bo);
+    if (planeCount != 1) {
+        LOG_ERROR("[MemoryMappedGPUBuffer] No support for multi-plane buffer objects. Linear, single plane buffers are necessary.");
+        return false;
+    }
+
+    auto dmaBufFD = gbm_bo_get_fd_for_plane(m_bo, 0);
+    if (dmaBufFD < 0) {
+        LOG_ERROR("[MemoryMappedGPUBuffer] Cannot access primary plane dma-buf.");
+        return false;
+    }
+
+    Vector<EGLAttrib> attributes = {
+        EGL_WIDTH, static_cast<EGLAttrib>(gbm_bo_get_width(m_bo)),
+        EGL_HEIGHT, static_cast<EGLAttrib>(gbm_bo_get_height(m_bo)),
+        EGL_LINUX_DRM_FOURCC_EXT, static_cast<EGLAttrib>(gbm_bo_get_format(m_bo))
+    };
+
+    std::array<EGLAttrib, 6> planeAttributes {
+        EGL_DMA_BUF_PLANE0_FD_EXT, static_cast<EGLAttrib>(dmaBufFD),
+        EGL_DMA_BUF_PLANE0_OFFSET_EXT, static_cast<EGLAttrib>(gbm_bo_get_offset(m_bo, 0)),
+        EGL_DMA_BUF_PLANE0_PITCH_EXT, static_cast<EGLAttrib>(gbm_bo_get_stride_for_plane(m_bo, 0))
+    };
+    attributes.append(std::span<const EGLAttrib> { planeAttributes });
+
+    if (m_modifier != DRM_FORMAT_MOD_INVALID) {
+        std::array<EGLAttrib, 4> modifierAttributes {
+            EGL_DMA_BUF_PLANE0_MODIFIER_HI_EXT, static_cast<EGLAttrib>(m_modifier >> 32),
+            EGL_DMA_BUF_PLANE0_MODIFIER_LO_EXT, static_cast<EGLAttrib>(m_modifier & 0xffffffff)
+        };
+        attributes.append(std::span<const EGLAttrib> { modifierAttributes });
+    }
+
+    attributes.append(EGL_NONE);
+
+    auto& display = WebCore::PlatformDisplay::sharedDisplay();
+    m_eglImage = display.createEGLImage(EGL_NO_CONTEXT, EGL_LINUX_DMA_BUF_EXT, nullptr, attributes);
+
+    close(dmaBufFD);
+    return !!m_eglImage;
+}
+
+bool MemoryMappedGPUBuffer::exportGBMBufferObjectAsDMABuf()
+{
+    if (auto dmaBufFD = gbm_bo_get_fd_for_plane(m_bo, 0); dmaBufFD >= 0) {
+        m_dmaBufFD = dmaBufFD;
```

Comment:
same -- we could use the UnixFileDescriptor and then we wouldn't need to close it explicitly in the destructor

Reply:
Nowadays using `RefPtr<DMABufBuffer> m_dmaBuf` -- comment is no longer applicable, and you were right of course :-)